api-reference/_hidden/dxDataGridColumn/calculateGroupValue.md

```diff
@@ -13,9 +13,295 @@ The current row's data.
+---
+##### jQuery
+
+    <!--JavaScript-->$(function() {
+        $("#{widgetName}Container").dx{WidgetName}({
+            columns: [
+                {
+                    dataField: "Country",
+                    groupIndex: 0,
+                    calculateGroupValue: "groupDataField"
+                },
+                "groupDataField"
+            ]
+        });
+    });
+
+##### Angular
+    
+    <!--HTML-->
+    <dx-{widget-name} ... >
+        <dxi-column
+            dataField="Country"
+            [groupIndex]="0"
+            calculateGroupValue="groupDataField">
+        </dxi-column>
+        <dxi-column
+            dataField="groupDataField"
+        </dxi-column>
+    </dx-{widget-name}>
+
+    <!--TypeScript-->
+    import { Dx{WidgetName}Module } from "devextreme-angular";
+    // ...
+    export class AppComponent {
+        // ...
+    }
+    @NgModule({
+        imports: [
+            // ...
+            Dx{WidgetName}Module
+        ],
+        // ...
+    })
+    
+
+##### Vue
+
+    <!-- tab: App.vue -->
+    <template>
+        <DxDataGrid ... >
+            <DxColumn
+                data-field="Position"
+                :group-index="0"
+                calculate-group-value="groupDataField"
+            />
+            <DxColumn
+                data-field="groupDataField"
+            />
+        </DxDataGrid>
+    </template>
+    <script>
+    import { DxDataGrid, DxColumn } from 'devextreme-vue/data-grid';
+
+    export default {
+        components: {
+            DxDataGrid,
+            DxColumn
+        },
+        // ...
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+    import DataGrid, { Column } from 'devextreme-react/data-grid';
+
+    const GroupCell = options => <div>{options.value}</div>;
+
+    function App() {
+        // ...
+        return (
+            <DataGrid ...>
+                <Column
+                    dataField="Position"
+                    groupIndex={0}
+                    calculateGroupValue="groupDataField"
+                />
+                <Column
+                    dataField="groupDataField"
+                />
+            </DataGrid>
+        );
+    }
+
+    export default App();
+
+---
+
+... or a function that creates custom groups:
+
+
+---
+##### jQuery
+
+    <!--JavaScript-->$(function() {
+        $("#{widgetName}Container").dx{WidgetName}({
+            columns: [
+                // ...
+                {
+                    dataField: "HireDate",
+                    dataType: "date",
+                    groupIndex: 0,
+                    calculateGroupValue: function(rowData){
+                        const convertDateToString = date => {
```

Comment:
Shouldn't this function be declared outside?

Reply:
No, although it can. It will be placed before the DataGrid container and this looks worse than the current version.